### PR TITLE
feat: adding the landingUrl config option

### PR DESCRIPTION
**Pull request by roja079**

### CODE REVIEW

File: src/messages/models/Modal/index.js [hunk 1]
```diff
@@ -325,19 +325,33 @@ export default {
...
-
                 if (target.tagName === 'IMG' && target.parentNode.tagName === 'A') {
-                    window.open(
-                        target.parentNode.href,
-                        'PayPal Credit Terms',
-                        'width=650,height=600,scrollbars=yes,resizable=no,location=no,toolbar=no,menubar=no,dependent=no,dialog=yes,minimizable=no'
-                    );
-
-                    evt.preventDefault();
+                    if (options.landingurl !== undefined && options.landingurl !== null) {
```
Comment (gillycheesesteak) [inline src/messages/models/Modal/index.js]: This condition will never be true, because a legacy NI banner (the condition above) will never have a landing URL

File: src/messages/models/Modal/index.js [hunk 2]
```diff
@@ -325,19 +325,33 @@ export default {
...
+                        evt.preventDefault();
+                    }
                 } else {
                     window.open(meta.clickUrl, '_blank');
                 }
             });
+        } else if (
+            options.style.layout === 'custom' &&
+            options.landingurl !== undefined &&
+            options.landingurl !== null
```
Comment (gillycheesesteak) [inline src/messages/models/Modal/index.js]: You should be able to replace this with just `if (options.style.layout === 'custom' && options.landingurl')`

File: src/messages/models/Modal/index.js [hunk 3]
```diff
@@ -325,19 +325,33 @@ export default {
...
+        ) {
+            events.on('click', evt => {
+                const { target } = evt;
+                if (target.tagName === 'IFRAME' || target.tagName === 'H1') {
```
Comment (gillycheesesteak) [inline src/messages/models/Modal/index.js]: I don't think we want to validate the target at all, because the custom banner could have any kind of markup in it

File: src/messages/models/Template/mutations/ezp_any_eqz.js [hunk 4]
```diff
@@ -43,7 +45,7 @@ export default {
             {
                 logo: Logo.ALTERNATIVE.COLOR,
                 headline: {
-                    replace: [['APR', 'APR.']],
+                    replace: [['APR', 'APR']],
```
Comment (gillycheesesteak) [inline src/messages/models/Template/mutations/ezp_any_eqz.js]: This line does nothing, it can be removed. This applies to any case where the two values passed to replace are the same.

File: src/messages/models/Modal/index.js (hunk 1 above)
Comment (roja079) [reply src/messages/models/Modal/index.js]: After making changes to the code i forgot to remove the if condition. Removed now.

File: src/messages/models/Modal/index.js (hunk 2 above)
Comment (roja079) [reply src/messages/models/Modal/index.js]: Done

File: src/messages/models/Modal/index.js (hunk 3 above)
Comment (roja079) [reply src/messages/models/Modal/index.js]: Done

File: src/messages/models/Template/mutations/ezp_any_eqz.js (hunk 4 above)
Comment (roja079) [reply src/messages/models/Template/mutations/ezp_any_eqz.js]: Done

File: src/messages/models/Banner/index.js [hunk 5]
```diff
@@ -90,6 +90,7 @@ const Banner = {
                 ) // Promise<Object(options, logger, wrapper, events, markup, template, meta)>
                 .then(
                     pipe(
+                        passThrough,
```
Comment (gillycheesesteak) [inline src/messages/models/Banner/index.js]: What is the purpose of this line?

File: src/messages/models/Modal/index.js [hunk 6]
```diff
@@ -338,6 +337,11 @@ export default {
                     window.open(meta.clickUrl, '_blank');
                 }
             });
+        }
+        if (options.style.layout === 'custom' && options.landingurl) {
```
Comment (gillycheesesteak) [inline src/messages/models/Modal/index.js]: This needs an else from the previous condition

File: src/messages/models/Template/mutations/ezp_any_gtz.js [hunk 7]
```diff
@@ -24,6 +24,7 @@ export default {
                 messageWidth: [250, 1000],
                 logo: Logo.ALT_NO_PP.COLOR,
                 headline: {
+                    replace: [['months', 'months']],
```
Comment (gillycheesesteak) [inline src/messages/models/Template/mutations/ezp_any_gtz.js]: We can remove all of the `replace` rules where the two values are the same - they have no effect

File: src/messages/models/Banner/validateOptions.js [hunk 8]
```diff
@@ -152,7 +152,7 @@ function getValidStyleOptions(logger, options) {
  * @param {Object} options User options object
  * @returns {Object} Object containing only valid options
  */
-export default curry((logger, { account, amount, countryCode, style, offer, ...otherOptions }) => {
+export default curry((logger, { account, amount, countryCode, style, offer, landingurl, ...otherOptions }) => {
```
Comment (gillycheesesteak) [inline src/messages/models/Banner/validateOptions.js]: Can we rename this field to `landingUrl`?

File: src/messages/models/Banner/index.js (hunk 5 above)
Comment (roja079) [reply src/messages/models/Banner/index.js]: That was kept for testing purpose, removed now.

File: src/messages/models/Modal/index.js (hunk 6 above)
Comment (roja079) [reply src/messages/models/Modal/index.js]: Done

File: src/messages/models/Template/mutations/ezp_any_gtz.js (hunk 7 above)
Comment (roja079) [reply src/messages/models/Template/mutations/ezp_any_gtz.js]: Done

File: src/messages/models/Banner/validateOptions.js (hunk 8 above)
Comment (roja079) [reply src/messages/models/Banner/validateOptions.js]: Done

File: src/messages/models/Template/mutations/ni_non-us.js [hunk 9]
```diff
@@ -8,7 +8,7 @@ export default {
             {
                 styles: [basicMediaQuery(320)],
                 logo: Logo.PRIMARY.COLOR,
-                headline: ['xsmall', { tag: 'medium', br: ['months'], replace: [['99+', '99+.']] }],
+                headline: ['xsmall', { tag: 'medium', br: ['months'], replace: [['99+', '99+']] }],
```
Comment (gillycheesesteak) [inline src/messages/models/Template/mutations/ni_non-us.js]: The replace rule here has no effect and can be removed

File: src/messages/models/Template/mutations/ni_non-us.js [hunk 10]
```diff
@@ -49,7 +50,7 @@ export default {
             {
                 styles: [basicMediaQuery(570)],
                 logo: Logo.ALTERNATIVE.COLOR,
-                headline: ['xsmall', { tag: 'medium', br: ['months'], replace: [['99+', '99+.']] }]
+                headline: ['xsmall', { tag: 'medium', br: ['months'], replace: [['99+', '99+']] }]
```
Comment (gillycheesesteak) [inline src/messages/models/Template/mutations/ni_non-us.js]: Remove replace rule

File: banners/ezp_any_eqz.json [hunk 11]
```diff
@@ -6,11 +6,11 @@
     },
     "data": {
         "headline": [
-            ["Pay over ${term} months at 0% APR", ["default", "small"]],
+            ["Pay over ${term} months at 0% APR.", ["default", "small"]],
             ["Enjoy ${term} Easy Payments with No Interest.", ["legacy-small"]]
         ],
         "subHeadline": [
-            ["Choose PayPal Credit and select ${term} Easy Payments to pay over time", ["legacy-medium"]],
+            ["Choose PayPal Credit and select ${term} Easy Payments to pay over time.", ["legacy-medium"]],
```
Comment (gillycheesesteak) [inline banners/ezp_any_eqz.json]: I think we should leave all the legacy messages the same. We could update the replace rules to match, but since we don't expect custom banners to use these messages, I think it is easier to just revert the changes to the messages
